arch-design/delfin/PerfomanceMontoringDesign.md

```diff
@@ -1,314 +1,313 @@
+* Metric name bandwidth is rarely used , most of the platforms consider data rate as throughput (MB/s) and IOPS as requests or iorate 
+* Common metric names are derived based on their availability in other SRM platforms 
+
+
+###### Common metric list
+
+**Device**
+
+| Delfin Metric  | Unit | Description                                                         |
+| -------------- | ---- | ------------------------------------------------------------------ |
+| throughput     | MB/s | Represents how much data is successfully transferred in MB/s       |
+| responseTime   | ms   | Average time taken for an IO operation in ms                       |
+| requests       | IOPS | Input/output operations per second                                 |
+| readThroughput | MB/s | Represents how much data read is successfully transferred in MB/s  |
+| writeThrouput  | MB/s | Represents how much data write is successfully transferred in MB/s |
+| readRequests   | IOPS | Read requests per second                                           |
+| writeRequests  | IOPS | Write requests per second                                          |
+| memoryUsage    | %    | Internal memory usage in %                                         |
+|                |      |                                                                    |
 
-```
-```
-keys = ['response_time', 'throughtput', 'bandwidth', 'read_throughtput', 'write_throughtput', 'read_bandwidth', 'write_bandwidth']
-```
-```
-@abc.abstractmethod
-def get_storage_perf_metrics(self, context, collection_spec, keys):
-    """Get storage device performance metrics  
-   Return array of metrics metrics[]"""
-    pass
+**Storage Pool**
 
-```
+| Metric Name     | Unit | Description                                                        |
+| --------------- | ---- | ------------------------------------------------------------------ |
+| Throughput      | MB/s | Represents how much data is successfully transferred in MB/s       |
+| responseTime    | ms   | Average time taken for an IO operation in ms                       |
+| Requests        | IOPS | Input/output operations per second                                 |
+| readThroughput  | MB/s | Represents how much data read is successfully transferred in MB/s  |
+| writeThroughput | MB/s | Represents how much data write is successfully transferred in MB/s |
+| readRequests    | IOPS | Read requests per second                                           |
+| writeRequests   | IOPS | Write requests per second                                          |
+
+**Volume**
+
+| Delfin Metric     | unit | Descrition                                                         |
+| ----------------- | ---- | ------------------------------------------------------------------ |
+| throughput        | MB/s | Represents how much data is successfully transferred in MB/s       |
+| responseTime      | ms   | Average time taken for an IO operation in ms                       |
```

Comment:
Is it latency?

Reply:
though there is theoratical difference , most of the platforms give only responsetime. latency is mostly used from a host perpective. Again if driver is mapping latency to reponse time, native_name and description should be given in the label.